lib/HAL/Compile/ThreadedCompilationManager.cpp

```diff
@@ -45,8 +47,9 @@ using namespace qssc::hal::compile;
 
 ThreadedCompilationManager::ThreadedCompilationManager(
     qssc::hal::TargetSystem &target, mlir::MLIRContext *context,
+    const config::QSSConfig &config, const OptDiagnosticCallback &diagnosticCb,
     ThreadedCompilationManager::PMBuilder pmBuilder)
-    : TargetCompilationManager(target, context),
+    : TargetCompilationManager(target, context, config, diagnosticCb),
```

Comment:
What I suggest above has the added benefit of removing the config/diagnostic callback from the target compilation manager.

Reply:
Removed in https://github.com/openqasm/qe-compiler/pull/240/commits/07c5019bb53affc143bef6f8ff87faf93c42426d